packages/main/src/plugin/context/context.ts

```diff
@@ -0,0 +1,104 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import type { ApiSenderType } from '../api.js';
+import type { IContext } from '../api/context-info.js';
+
+export class Context implements IContext {
+  private _value: Record<string, any>;
+
+  constructor(
+    private _id: number,
+    private _parent: Context | null,
+    private _extension: string | null,
+    private apiSender: ApiSenderType,
+  ) {
+    this._value = Object.create(null);
+    this._value['_contextId'] = _id;
```

Comment:
do we need that attribute ? not sure as you delete it when collecting all values ?

Reply:
I took it from vscode. I left it as i'm not sure what could be the need for it honestly. But i can remove it and we can add it again in future if we see a valid reason

Reply:
what is strange to me is that the _id is based from the length of contexts

you're doing

```
new Context(this.contexts.length
```

so it means two contexts can have the same id (if you add and then remove a context, length will be reduced)

Reply:
ah you're right, good catch